### PR TITLE
Add method to compute induced norm of SecondQuantizedOp

### PR DESCRIPTION
**Pull request by kevinsung**
### Summary
Adds `SecondQuantizedOp.induced_norm`.

### CODE REVIEW

**Review by mrossinek — CHANGES_REQUESTED** [review]
The same comment as in the other PR applies: after we merged #746, please re-apply these changes on the new code location (`qiskit_nature/second_q/operators`) and leave the old location untouched.

Besides that, please also provide a reference for this definition/implementation of the induced p-norm. I seem to not be able to find a matching definition and would like to verify the naming.

**Comment by kevinsung** [comment]
For the reference, would https://en.wikipedia.org/wiki/Norm_(mathematics)#p-norm be okay? This is the p-norm induced by the coefficients of the operator, hence the name. If the operator basis is orthonormal then this is a standard definition, so just `norm` or `p_norm` would be fine. The somewhat non-standard behavior arises for FermionicOps that are not normal-ordered, since the norm may change after normal-ordering. This is the reason to call it `induced_norm` instead of `norm` to make it clear that it is the norm induced by the coefficients. We could choose to always normal-order first, but it's better to give the user the option not to do that.

File: qiskit_nature/operators/second_quantization/second_quantized_op.py [hunk 1]
```diff
@@ -65,3 +65,24 @@ def is_hermitian(self) -> bool:
         return frozenset(self.simplify().to_list()) == frozenset(
             self.adjoint().simplify().to_list()
         )
+
+    def induced_norm(self, order: int = 1) -> float:
+        r"""Returns the induced p-norm of the operator.
```
Comment (mrossinek) [inline qiskit_nature/operators/second_quantization/second_quantized_op.py:70]: I suggest that you add the explanations which you gave in the comment on this PR above in this docstring.
You could even link to the p-norm definition on Wikipedia

**Review by mrossinek — COMMENTED** [review]
OIther than this simple comment, this LGTM :+1:

File: qiskit_nature/operators/second_quantization/second_quantized_op.py (hunk 1 above)
Comment (kevinsung) [reply qiskit_nature/operators/second_quantization/second_quantized_op.py]: Done.

**Comment by kevinsung** [comment]
@mrossinek I've updated this PR for the library refactor and addressed your comment, it should be good to go.

**Review by mrossinek — APPROVED** [review]
LGTM, thanks! :+1:

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**